### ISSUE TITLE
Hi-Lander : How to set level number

### ISSUE THREAD

**Issue by ZixiangPan1997** [dmlc/dgl#3945]
## ❓ Questions and Help

Hi, when I try to reimplement your paper "Learning Hierarchical Graph Neural Networks for Image Clustering"
https://arxiv.org/abs/2107.01319v1
Tabel 2 and Tabel 3 all obtained very high perfermance compared to your benchmark methods. But whether these performance on those dataset used save parameter of (k,l and Pτ)？
Thank you

Before proceeding, please note that we recommend
using our discussion forum (https://discuss.dgl.ai) for
general questions. As a result, this issue will
likely be CLOSED shortly.

**Comment by hetong007** [comment]
Hi, thanks for your interests in DGL and HiLANDER. The hyperparameters for Table 2 and Table 3 are specified in Section 4.2, i.e. Implementation Details. In terms of the number of layers in training, if not described in paper, I’d suggest you to try L=3. When doing inference, the number of layers is determined by early-stopping in Section 4.3.

**Comment by ZixiangPan1997** [comment]
Thank you for your reply. There's another question about your model Hi-Lander:
As were mentioned in the paper, two innovation contribution points are Hirachical Clustering and Meta-Learning.
My question is : in train_subg.py and test_subg.py, how did you train the "meta" in the pipeline? There's no clear evidence that loss function calculated gradient on two parameters k and l?
Thank you
With best regards


Pan
2022-5-7
&nbsp;
&nbsp;
------------------&nbsp;Original&nbsp;------------------
From: ***@***.***&gt;;
Date: &nbsp;Sun, Apr 24, 2022 04:21 PM
To: ***@***.***&gt;; 
Cc: ***@***.***&gt;; ***@***.***&gt;; 
Subject: &nbsp;Re: [dmlc/dgl] Hi-Lander : How to set level number (Issue #3945)

&nbsp;



 
Hi, thanks for your interests in DGL and HiLANDER. The hyperparameters for Table 2 and Table 3 are specified in Section 4.2, i.e. Implementation Details. In terms of the number of layers in training, if not described in paper, I’d suggest you to try L=3. When doing inference, the number of layers is determined by early-stopping in Section 4.3.

**Comment by hetong007** [comment]
The two parameters, `k` and `l`, are used in the dataset class: https://github.com/dmlc/dgl/blob/master/examples/pytorch/hilander/dataset/dataset.py#L12

Specifically `k` affects the constructed graph in each level, and `l` controls the number of levels. The above dataset class creates a dataset instance that includes `k`-nn graphs from all `l` levels.

**Comment by ZixiangPan1997** [comment]
Thank you for your instant and patient reply.
The problem confused me is that did Hi-lander really utilized meta-learning?
If so, for example, according to tabel 2 &amp; 3, Could you please tell me how do you arrange support/query task set? And how do Hi-Lander implement meta-learning "Learn to learn" (namely tuning model parameter (k,l) via support task set training and query task set testing)? When the method meet with an unseen task , how do hi-lander generate a new model (merely "where is the 'meta' you trained") for the task?
Thank you


With Best Regards


Pan
2022-5-8
&nbsp;
------------------&nbsp;Original&nbsp;------------------
From: ***@***.***&gt;;
Date: &nbsp;Sat, May 7, 2022 06:09 PM
To: ***@***.***&gt;; 
Cc: ***@***.***&gt;; ***@***.***&gt;; 
Subject: &nbsp;Re: [dmlc/dgl] Hi-Lander : How to set level number (Issue #3945)

&nbsp;



 
The two parameters, k and l, are used in the dataset class: https://github.com/dmlc/dgl/blob/master/examples/pytorch/hilander/dataset/dataset.py#L12
 
Specifically k affects the constructed graph in each level, and l controls the number of levels. The above dataset class creates a dataset instance that includes k-nn graphs from all l levels.
